cli/azd/pkg/stringutil/stringutil.go

```diff
@@ -0,0 +1,13 @@
+package stringutil
```

Comment:
Can we start by adding this functions to the `infra` package only as non-exported?

Reply:
If we are to make this generic I would just rename the package to be `compare` and keep them as exported.  I don't see any harm in exporting a general purpose package that we can then adopt in other places later.